### PR TITLE
Add tag tracking

### PR DESCRIPTION
**Pull request by brannow**
Allows to gather tag information

### CODE REVIEW

File: src/GitWrapper/GitRepository.php [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 class GitRepository
 {
+    const TAG_ORIGIN = 'tags';
```
Comment (IchHabRecht) [inline src/GitWrapper/GitRepository.php]: As this doesn't belong to the function, I would kindly ask you to remove the unnecessary constant.

**Review by IchHabRecht — CHANGES_REQUESTED** [review]

**Comment by IchHabRecht** [comment]
Thanks for this pull request. Nearly perfect now, but a test like "testTagReturnsTags" is missing. Would you mind to add it?

File: tests/GitWrapper/GitRepositoryTest.php [hunk 2]
```diff
@@ -176,6 +176,29 @@ public function testHasChangesFindsChanges()
...
+            $this->gitWrapper->execute('tag', [['a' => $tagString], ['m' => 'Tag Version']], [], $repositoryDirectory);
+        }
+
+        $this->assertSame(
+            [
+                'v1.4.1-rc_34123',
+                '1.0.1',
+                '0.3.1-test',
+            ],
+            array_reverse($localRepository->tag(['l']))
```
Comment (IchHabRecht) [inline tests/GitWrapper/GitRepositoryTest.php]: why do you use array_reverse here? you could define $tags in the expected order already and re-use it for your assertion

File: tests/GitWrapper/GitRepositoryTest.php [hunk 3]
```diff
@@ -176,6 +176,29 @@ public function testHasChangesFindsChanges()
         $this->assertTrue($this->gitRemoteRepository->hasChanges());
     }
 
+    public function testTagReturnsTags()
+    {
+        $localRepository = $this->initializeLocalRepository();
+        $repositoryDirectory = rtrim($localRepository->getDirectory(), '/\\') . '/';
+        $tagStrings = [
```
Comment (IchHabRecht) [inline tests/GitWrapper/GitRepositoryTest.php]: please rename to $tags

**Review by IchHabRecht — CHANGES_REQUESTED** [review]

File: tests/GitWrapper/GitRepositoryTest.php (hunk 2 above)
Comment (brannow) [reply tests/GitWrapper/GitRepositoryTest.php]: `git tag -l` reorder the tags itself. 
the reverse was simply to be compensate the missing option in git under 2.0. 
yes, It not really necessary.

**Pull request merged**